### ISSUE TITLE
Add smoke test for macOS glue code

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#4287]
Panicks like #4281 and #4282 (and others) could have been caught by a simple crate similar to `firezone-tunnel` that runs on the macOS runner and just calls `Session.connect` from inside a main function to try bringing up the runtime.

We can create the TUN device manually first, or modify Device to have a code path for creating a TUN conditionally.

Inside the NetworkExtension, which we have to use to be on the App Store, we're not allowed to create TUN devices. But running on "bare metal" on the macOS runner won't have that limitation.

cc @ReactorScram

**Comment by ReactorScram** [comment]
Okay so 🤔 If this is all in 1 process, and doesn't need special Swift / NetworkExtension stuff, it could theoretically be done in Rust, maybe even inside `cargo test`?

Might overlap with #4254. macOS does use sudo so maybe the same thing will work as on Linux, to have certain Cargo unit tests run with sudo powers.

**Comment by jamilbk** [comment]
@ReactorScram Yeah, this would essentially be a headless macOS client test, so one process (not an actual headless client we could ship, but you get the idea).

The thing is, we don't create the tun device in the actual client, Apple does, and we just find it by walking the process' file descriptor table looking for the first `utun` device we have access to.

So for this to be a realistic test we'd have to create the TUN device with the same properties Apple's Network Extension framework does.

The other route which is probably a better investment long-term is to start mocking these TUN devices with a realistic stub so we can simulate any real-world condition we'd like, but that's probably out of scope for this issue.
